src/python/dxpy/nextflow/nextflow_templates.py

```diff
@@ -78,7 +78,11 @@ def get_nextflow_src(custom_inputs=None, profile=None, resources_dir=None):
             '''% (i['name'], i['name'], value)
         # required inputs need to be added as runtime pipeline params
         else:
-            required_runtime_params += " --{}={}".format(i["name"], value)
+            required_runtime_params = required_runtime_params + f'''
+            if [ -n "${i['name']}" ]; then
+                required_inputs="${{required_inputs}} --{i['name']}=${i['name']}"
+            fi
+            '''
```

Comment:
f string might not be compatible with python 2

Reply:
The pipeline applets are always Ubuntu 20.04+python3 though so IMHO we can leave the f string.